src/Microsoft.Data/DataFrame.cs

```diff
@@ -207,6 +207,19 @@ public IList<string> Columns
             }
         }
 
+        public IList<string> ColumnTypes
```

Comment:
I don't think this is the right API to expose to people. We shouldn't be dealing with types as strings.

Also, we should have some class/struct that represents the Schema that contains columns with names and types, and potentially other information.

Reply:
Also, returning a new List on every call to a property is an anti-pattern IMO.

---
In reply to: [327715603](https://github.com/dotnet/corefxlab/pull/2730#discussion_r327715603) [](ancestors = 327715603)